givitsite/gatherer/agora.py

```diff
@@ -0,0 +1,137 @@
+    return items_list
+
+
+def get_iseek(item):
+    """
+    this function gets name item in hebrew and return the iseek code in agora website.
+    Parameters:
+    item (string): item is the name of our item in hebrew.
+    Returns:
+    iseek (string): represent the code of our current item in the url, its in a dict such that iseek is the value of key item.
+    """
+    iseek_dict = {'ארון': '20009', 'מיטה': '20016', 'כיסא': '20008',
+                  'מקרר': '10006', 'מכונת כביסה': '10029', 'ספה': '20017'}
+    return iseek_dict[item]
+
+
+def Agora_Getrequest(iseek):
+    """
+    this function gets iseek which allow us to nevigate to the specific url and return beatifulsoup object.
+    Parameters:
+    iseek (string): iseek is the category number of the product that we need for the url.
+    Returns:
+    BeautifulSoup object: which represents the document as a nested data structure.
+    """
+    url = 'https://www.agora.co.il/toGet.asp?'
+    headers = {}
+    headers['User-Agent'] = 'Mozilla/5.0'
+    values = {'searchType': 'subCategory', 'dealType': '1', 'iseek': iseek}
+    data = urllib.parse.urlencode(values)
+    new_url = url + data
+    data = data.encode('utf-8')
+    req = urllib.request.Request(new_url, data, headers)
+    webpage = urllib.request.urlopen(req).read()
+    soup = BeautifulSoup(webpage, 'html.parser', from_encoding="utf-8")
+    return soup
+
+
+def find_furniture(soup, area, name):
+    """
+    this function gets soup, area and name of the product we are looking for
+    and return all the rows that found on the website as url_list.
+    Parameters:
+    soup (beautifulsoup object): which represents the document as a nested data structure.
+    area (string): the area that we want to pick the product from.
+    name (string): the name of the product we search for.
+    Returns:
+    url_list (list): list of all url that match to what we search.
+    """
+    url_list = []
+    t = soup.findAll('tbody', class_='objectGroup')
+    for i in range(len(t)):
+        c1 = t[i].find_all('td', class_='area')
+        c2 = t[i].find_all('td', class_='newWindow')
+        c5 = t[i].find_all('td', class_='objectName')
+        c6 = t[i].find_all('td', class_='objectState')
+        item_name = str(c5[0].text)
+        check_item_state = (str((c6[0].select('span'))))
+        item_state = str(re.match('.*(condition[2,1]).*', check_item_state))
+        if area in (c1[0].text) and str(re.match(name, item_name)) != 'None' and item_state != 'None':
+            c3 = c2[0].find_all('a')
+            string1 = str(c3[0])
+            string2 = str(re.findall('href="(\S*)"', string1))
+            string4 = 'https://www.agora.co.il/' + string2[2:-2]
+            url_list.append(string4)
+    return url_list
+
+
+def find_new_furniture(soup, area, name):
+    """
+    this function is the same as find_furniture, the only diffrent is that here we search only item that add this day.
```

Comment:
Since this and `find_furniture` are so similar, you might want to refactor them so you can mostly share the common code. For example - have a common function that breaks down the HTML into a data structure that is easier to use for what you need and then have ` find_furniture` and `find_new_furniture` be smaller functions that use it to extract the pieces they need.

That would probably also make it easier to write tests later.